### PR TITLE
bionic-prebuild: Export static libc, libm, libdl as well

### PR DESCRIPTION
**Pull request by angerman**
Building static android (linux-bionic) binaries requires us to have access to static versions of libc, libm, libdl, ... Export those as well.

### CODE REVIEW

**Comment by Ericson2314** [comment]
```nix
, enableStatic ? stdenv.hostPlatform.isStatic
, enableShared ? !stdenv.hostPlatform.isStatic
```
at the top would be good

File: pkgs/os-specific/linux/bionic-prebuilt/default.nix [hunk 1]
```diff
@@ -92,9 +94,18 @@ stdenvNoCC.mkDerivation rec {
     cp -v ${prebuilt_crt.out}/*.o $out/lib/
     cp -v ${prebuilt_crt.out}/libgcc.a $out/lib/
     cp -v ${prebuilt_ndk_crt.out}/*.o $out/lib/
+    ${if enableShared then ''
     for i in libc.so libm.so libdl.so liblog.so; do
       cp -v ${prebuilt_libs.out}/$i $out/lib/
     done
+    ''}
```
Comment (Ericson2314) [inline pkgs/os-specific/linux/bionic-prebuilt/default.nix:101]:
```suggestion
    '' + lib.optionalString enableShared ''
    for i in libc.so libm.so libdl.so liblog.so; do
      cp -v ${prebuilt_libs.out}/$i $out/lib/
    done
    '' + ''
```

File: pkgs/os-specific/linux/bionic-prebuilt/default.nix [hunk 2]
```diff
@@ -92,9 +94,18 @@ stdenvNoCC.mkDerivation rec {
...
+    ${if enableStatic then ''
+    # no liblog.a; while it's also part of the base libraries,
+    # it's only available as shared object in the prebuilts.
+    for i in libc.a libm.a libdl.a; do
+      cp -v ${prebuilt_ndk_crt.out}/$i $out/lib/
+    done
+    ''}
```
Comment (Ericson2314) [inline pkgs/os-specific/linux/bionic-prebuilt/default.nix:108]:
```suggestion
     '' + lib.optionalString enableStatic ''
    # no liblog.a; while it's also part of the base libraries,
    # it's only available as shared object in the prebuilts.
    for i in libc.a libm.a libdl.a; do
      cp -v ${prebuilt_ndk_crt.out}/$i $out/lib/
    done
    '' + ''
```

**Review by Ericson2314 — APPROVED** [review]

**Comment by angerman** [comment]
@Ericson2314 this one also needs a backport to 21.11

**Review by s1341 — COMMENTED** [review]
lgtm!

**Pull request merged**